src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/SortOrderQueryBuilder.java

```diff
@@ -12,23 +21,39 @@
     private static final String SPACE = " ";
     private static final String COMMA = ",";
 
+    private static final Set<String> VALID_COLUMNS_FOR_ORDER_BY
+        = Collections.unmodifiableSet((Set<? extends String>) Stream
+                                          .of(CaseDetailsEntity.CREATED_DATE_FIELD_COL,
+                                              CaseDetailsEntity.LAST_MODIFIED_FIELD_COL,
+                                              CaseDetailsEntity.JURISDICTION_FIELD_COL,
+                                              CaseDetailsEntity.CASE_TYPE_ID_FIELD_COL,
+                                              CaseDetailsEntity.STATE_FIELD_COL,
+                                              CaseDetailsEntity.REFERENCE_FIELD_COL,
+                                              CaseDetailsEntity.SECURITY_CLASSIFICATION_FIELD_COL)
+                                          .collect(Collectors.toCollection(HashSet::new)));
+
     public String buildSortOrderClause(MetaData metaData) {
         StringBuilder sb = new StringBuilder();
         metaData.getSortOrderFields().forEach(sortOrderField -> {
-            if (sortOrderField.isMetadata()) {
+            if (sortOrderField.isMetadata() && VALID_COLUMNS_FOR_ORDER_BY.contains(getMataFieldName(sortOrderField.getCaseFieldId()))) {
```

Comment:
this validation already occurs in definition-store. If we re-force it here then we should throw an error immediately rather than fallback to JSONB group. 

Also `VALID_FIELD_NAME_REGEX` should in sync with fieldNames pattern of definition.